### DISCUSSION TITLE
Option to enable iscisd even when enable_longhorn=false in kube.tf?

### DISCUSSION THREAD

**Discussion by valkenburg-prevue-ch** [mysticaltech/terraform-hcloud-kube-hetzner#1172] [Ideas]
The line "systemctl enable --now iscsid" in https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/34ecefd0834e050f76d7ae058e7a2319db4e16bf/agents.tf#L108 is conditional on longhorn being enabled. But some of the users install longhorn separately outside of kube.tf, and there might be users who want iscsid for whatever reason.

Can we enable iscsid when longhorn is not installed by kube.tf? I see two options:
1. remove the conditional, just always enable iscsid. (less complexity, but iscsid running also when you do not want it)
2. add a new variable which enables iscsid even when enable_longhorn is false.

Any thoughts?

Note, for searchability: I encountered the need for this line, when kured rebooted my nodes and suddenly longhorn volumes could not attach anymore. The volumes where constantly spinning as 'attaching', repeatedly restarting the pods which held the claims for these volumes. After digging, I found that the logs of longhorn manager showed what was wrong: "Wait for gRPC service of process pvc to start", after which I understood that the iscsid never restarted after the system reboot. Because it was probably once started, but not enabled by systemctl (as per the tf line linked above).

**Comment by mysticaltech** [comment]
This is now enabled by default.
